Modules/selectmodule.c

```diff
@@ -554,9 +555,17 @@ poll_poll(pollObject *self, PyObject *args)
             return NULL;
         }
 
-        deadline = _PyTime_GetMonotonicClock() + timeout;
+        if (timeout >= 0)
+            deadline = _PyTime_GetMonotonicClock() + timeout;
     }
 
+    if (ms < 0) /* On many OSes timeout must be INFTIM or -1, issue 31334 */
```

Comment:
Style nit: Same as above.

Reply:
Can we make this comment clearer? For example, it would be nice to mention this is the case if negative timeout is less than -1 on some OSes.

Also, I'd replace "many" with "some" or "some BSD OSes" since this is the first time this issue has been reported.